app/Models/Payment.php

```diff
@@ -134,4 +139,29 @@ public function getRelationshipPermissionMap(): array
             'payable' => 'dues-transactions',
         ];
     }
+
+    public function updateFromSquareCashTransaction(SquareCashTransaction $transaction): void
+    {
+        $this->amount = $transaction->amount;
+        $this->processing_fee = null;
+        $this->method = 'squarecash';
+        $this->square_cash_transaction_id = $transaction->transaction_id;
+        $this->created_at = $transaction->transaction_timestamp;
+        $this->save();
+    }
+
+    public function updateFromSquareTransaction(SquareTransaction $transaction): void
+    {
+        $this->method = 'swipe';  // note that some POS transactions are actually keyed for some reason
+        $this->amount = $transaction->amount;
+        $this->processing_fee = $transaction->processing_fee;
+        $this->created_at = $transaction->transaction_timestamp;
```

Comment:
It occured to me that we might want to add a `transaction_timestamp` to the `Payment` model in lieu of backdating these fields. That would also help with #830 somewhat.

Reply:
Agreed that this would be good eventually but it doesn't need to happen for this PR.